supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/callcaching/AwsBatchBackendCacheHitCopyingActor.scala

```diff
@@ -47,53 +49,59 @@ import scala.language.postfixOps
-        (newDetritus + (JobPaths.CallRootPathKey -> destinationCallRootPath)) -> newDetritus.values.map(S3BatchCommandBuilder.touchCommand).toSet
+        // Don't forget to re-add the CallRootPathKey that has been filtered out by detritusFileKeys
+        TryUtil.sequenceMap(detritusAsPaths, "Failed to make paths out of job detritus") map { newDetritus =>
+          (newDetritus + (JobPaths.CallRootPathKey -> destinationCallRootPath)) -> newDetritus.values.map(S3BatchCommandBuilder.touchCommand).toSet
+        }
       }
-  }
+      case (_, _) => super.processDetritus(sourceJobDetritusFiles)
+   }
 
   override protected def additionalIoCommands(sourceCallRootPath: Path,
                                               originalSimpletons: Seq[WomValueSimpleton],
                                               newOutputs: CallOutputs,
                                               originalDetritus:  Map[String, String],
-                                              newDetritus: Map[String, Path]): List[Set[IoCommand[_]]] = {
-    cachingStrategy match {
-      case UseOriginalCachedOutputs =>
-        val content =
-          s"""
-             |This directory does not contain any output files because this job matched an identical job that was previously run, thus it was a cache-hit.
-             |Cromwell is configured to not copy outputs during call caching. To change this, edit the filesystems.aws.caching.duplication-strategy field in your backend configuration.
-             |The original outputs can be found at this location: ${sourceCallRootPath.pathAsString}
+                                              newDetritus: Map[String, Path]): List[Set[IoCommand[_]]] = (batchAttributes.fileSystem, cachingStrategy)  match {
+      case (AWSBatchStorageSystems.s3, UseOriginalCachedOutputs) =>
+            val content =
+              s"""
+                 |This directory does not contain any output files because this job matched an identical job that was previously run, thus it was a cache-hit.
+                 |Cromwell is configured to not copy outputs during call caching. To change this, edit the filesystems.aws.caching.duplication-strategy field in your backend configuration.
+                 |The original outputs can be found at this location: ${sourceCallRootPath.pathAsString}
       """.stripMargin
 
-        List(Set(S3BatchCommandBuilder.writeCommand(jobPaths.callExecutionRoot / "call_caching_placeholder.txt", content, Seq(CloudStorageOptions.withMimeType("text/plain")))))
-      case CopyCachedOutputs => List.empty
+            List(Set(S3BatchCommandBuilder.writeCommand(jobPaths.callExecutionRoot / "call_caching_placeholder.txt", content, Seq(CloudStorageOptions.withMimeType("text/plain")))))
+       case (AWSBatchStorageSystems.s3, CopyCachedOutputs) => List.empty
+       case (_, _) => super.additionalIoCommands(sourceCallRootPath,originalSimpletons, newOutputs, originalDetritus,newDetritus)
```

Comment:
Does this mean the EFS logic does not respect the distinction between `UseOriginalCachedOuputs` and `CopyCachedOutputs`?

Reply:
Yes. The EFS logic behaves the same way as shared file system.  UseOriginalCachedOuputs and CopyCachedOutputs distinction is used only  in the context of  object store --S3/AWS or Google.